packages/api/utils/validateRequestShape.test.ts

```diff
@@ -0,0 +1,37 @@
+import { validateRequestShape } from "../utils/validateRequestShape";
+import { ApiGatewayEventParsed } from "./eventHandlingTool";
+import { v4 as uuidv4 } from "uuid";
+import { Environment } from "../../orm/entity/Environment";
+
+describe("Shape validation tests", () => {
+  jest.spyOn(console, "error").mockImplementation(() => jest.fn()); // suppress output
+  const validEventRequest: ApiGatewayEventParsed<Environment> = {
+    body: { name: "env name" },
+    pathParameters: { portfolioId: uuidv4(), applicationId: uuidv4(), environmentId: uuidv4() },
+  } as any;
+  const undefinedBodyEvent: ApiGatewayEventParsed<Environment> = {
+    body: undefined,
+    pathParameters: { portfolioId: uuidv4() },
+  } as any;
+  const invalidUuidv4: ApiGatewayEventParsed<Environment> = {
+    body: { name: "env name" },
+    pathParameters: { portfolioId: uuidv4(), applicationId: "not uuidv4" },
+  } as any;
+
+  it("should return a SetupSuccess object", async () => {
```

Comment:
Maybe a couple of additional tests that check for `SetupSuccess` / `SetupError`
```suggestion
  it("should return a SetupSuccess object", async () => {
    expect(validateRequestShape<Environment>(validEventRequest)).toBeInstanceOf(SetupSuccess);
  });
  it("should return this exact object", async () => {
```

Reply:
Added an additional test for `SetupSuccess`. I do not think there are any `SetupError`, but the `.toThrow` catches the errors thrown when there is an error.